OKR.md

```diff
@@ -0,0 +1,50 @@
+# Quarterly Objectives and Key Results
+
+We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
+
+## 2018 Q4
+
+**go-ipfs handles large datasets (1TB++) without a sweat**
+- `PX` - It takes less than 48 hours to transfer 1TB dataset over Fast Ethernet (100Mbps)
+- `PX` - It takes less than 12 hours to transfer 200GB sharded dataset over Fast Ethernet (100Mbps)
+- `PX` - There is a prototype implementation of GraphSync for go-ipfs
+- `PX` - There is a better and more performant datastore module (e.g Badger or better)
+- `P1` - Rewrite pinning data structures to support large data sets / many files performantly
+
+**The bandwidth usage is reduced significantly and is well kept under control**
+- `PX` - Users can opt out of providing every IPLD node (and only provide root hashes)
+- `PX` - "Bitswap improvements around reducing chattiness, decreasing bandwidth usage (fewer dupe blocks), and increasing throughput"
+
+**It is a joy to use go-ipfs programatically**
+- `PX` @magik6k - The Core API is finalized and released. Make it easier to import go-ipfs as a package
+- `PX` - go-ipfs-api exposes the new Core API
+- `PX` - go-ipfs Daemon, Gateway, and cmds library use the new Core API
+- `PX` - The legacy non Core API is deprecated and the diagram on go-ipfs README is updated
+
+**go-ipfs becomes a well maintained project**
+- `PX` @eingenito - The Go Contributing Guidelines are updated to contemplate expecations from Core Devs and instructions on how to be an effective contributor (e.g include PR templates)
+- `PX` @eingenito - A Lead Maintainer Protocol equivalent is proposed, reviewed by the team, merged and implemented
+- `PX` @eingenito - Every issue on https://waffle.io/ipfs/go-ipfs gets triaged (reviewed and labeled following https://github.com/ipfs/pm/blob/master/GOLANG_CORE_DEV_MGMT.md)
+- `PX` - Every package has tests and tests+code coverage are running on Jenkins
+- `PX` - There is an up-to-date Architecture Diagram of the Go implementation of IPFS that links packages to subsystems to workflows
+
+**gx becomes a beloved tool by the Go Core Contributors**
+- `PX` - 
+- `PX` - 
+
+**Complete outstanding endeavours and still high priorities from Q3**
+- `P0` @kevina - base32 is supported and enabled by default
+- `PX` - go-ipfs gets a unixfsV2 prototype
```

Comment:
before a prototype we should probably have a spec we agree on =]

Reply:
Mikeal wants to hand off spec work to someone else

Reply:
@mikeal is working on this...

Reply:
To be more specific, I'm working on the spec, and I can easily hand off parts of the spec once the skeleton lands (should be done next week).

We should be able to start implementing once the (skeleton + data) PRs land, as we don't need special file types for MFS (as far as I know). I'll prioritize getting those landed so that we don't block early implementation work.

I'm not that familiar with the Go side of things but I believe @warpfork and @Stebalien mentioned that the IPLD API's on the Go side need to get revised and I'm not sure if that work is also going to block this implementation. If the goal of the unixfs-v2 implementation is to be somewhat codec agnostic then it probably will. Review and revisions to those APIs are in the IPLD Q4 OKR's.